### PR TITLE
FAT-1312:C15858 Add expense class to budget 

### PR DESCRIPTION
**Pull request by IhorBohdan**

### CODE REVIEW

**Review by kimkate303 — APPROVED** [review]

File: cypress/integration/finance/funds/add-expense-class-to-budget.spec.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import InteractorsTools from '../../../support/utils/interactorsTools';
+
+describe('ui-finance: Add budget to fund', () => {
+  const firstExpenseClass = { ...NewExpenceClass.defaultUiBatchGroup };
+  const fund = { ...NewFund.defaultFund };
+  const defaultLedger = NewLedger.defaultLedger;
+  const defaultFiscalYear = { ...NewFiscalYear.defaultFiscalYear };
+
+  before(() => {
+    cy.login(Cypress.env('diku_login'), Cypress.env('diku_password'));
```
Comment (penfriend) [inline cypress/integration/finance/funds/add-expense-class-to-budget.spec.js:24]: cy.loginAsAdmin

File: cypress/integration/finance/funds/add-expense-class-to-budget.spec.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+  it('C15858 Add expense class to budget (thunderjet)', { tags: [testType.criticalPath, devTeams.thunderjet] }, () => {
+    cy.visit(TopMenu.fundPath);
+    FinanceHelp.searchByName(fund.name);
+    FinanceHelp.selectFromResultsList();
+    Funds.seletBudgetDetails();
+    Funds.editBudget();
+    Funds.addExpensesClass(firstExpenseClass.name);
+    InteractorsTools.checkCalloutMessage(`Budget ${fund.code} has been saved`);
+    // Need to wait callout massage
```
Comment (penfriend) [inline cypress/integration/finance/funds/add-expense-class-to-budget.spec.js:63]: massage???

File: cypress/integration/finance/funds/add-expense-class-to-budget.spec.js (hunk 1 above)
Comment (IhorBohdan) [reply cypress/integration/finance/funds/add-expense-class-to-budget.spec.js]: Done

File: cypress/integration/finance/funds/add-expense-class-to-budget.spec.js (hunk 2 above)
Comment (IhorBohdan) [reply cypress/integration/finance/funds/add-expense-class-to-budget.spec.js]: message*sorry

File: cypress/integration/finance/funds/add-expense-class-to-budget.spec.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+describe('ui-finance: Add budget to fund', () => {
+  const firstExpenseClass = { ...NewExpenceClass.defaultUiBatchGroup };
+  const fund = { ...NewFund.defaultFund };
+  const defaultLedger = NewLedger.defaultLedger;
+  const defaultFiscalYear = { ...NewFiscalYear.defaultFiscalYear };
+
+  before(() => {
+    cy.loginAsAdmin();
+    cy.visit(`${SettingsMenu.expenseClassesPath}`);
```
Comment (penfriend) [inline cypress/integration/finance/funds/add-expense-class-to-budget.spec.js:25]: Do you need here this syntax `${}` ? It is obsolete here

File: cypress/support/fragments/finance/funds/funds.js [hunk 4]
```diff
@@ -262,5 +263,33 @@ export default {
     cy.do([
       Accordion({ id: 'currentBudget' }).find(MultiColumnListCell({ content: fundCode.concat('-', fiscalYear) })).click()
     ]);
+  },
+
+  seletBudgetDetails:(rowNumber = 0) => {
```
Comment (TetianaParanich) [inline cypress/support/fragments/finance/funds/funds.js:268]: should it be selectBudgetDetails name?

Comment (IhorBohdan) [reply cypress/support/fragments/finance/funds/funds.js:268]: Yeah,I open the budget information in the following tab

**Review by TetianaParanich — APPROVED** [review]

File: cypress/integration/finance/funds/add-expense-class-to-budget.spec.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  const defaultLedger = NewLedger.defaultLedger;
+  const defaultFiscalYear = { ...NewFiscalYear.defaultFiscalYear };
+
+  before(() => {
+    cy.loginAsAdmin();
+    cy.visit(SettingsMenu.expenseClassesPath);
+    SettingsFinance.createNewExpenseClass(firstExpenseClass);
+
+    cy.visit(TopMenu.fiscalYearPath);
+    FiscalYears.createDefaultFiscalYear(defaultFiscalYear);
```
Comment (penfriend) [inline cypress/integration/finance/funds/add-expense-class-to-budget.spec.js:29]: I don't see the removing of the fiscal year

File: cypress/support/fragments/finance/funds/funds.js (hunk 4 above)
Comment (penfriend) [reply cypress/support/fragments/finance/funds/funds.js:268]: Tetiana means that you missed a letter in the name of the function: seletBudgetDetails

Comment (IhorBohdan) [reply cypress/support/fragments/finance/funds/funds.js]: done

File: cypress/integration/finance/funds/add-expense-class-to-budget.spec.js (hunk 5 above)
Comment (IhorBohdan) [reply cypress/integration/finance/funds/add-expense-class-to-budget.spec.js:29]: Done

**Pull request merged**